app/src/main/java/com/example/bigowlapp/activity/SupervisedGroupListActivity.java

```diff
@@ -1,59 +1,103 @@
+
 public class SupervisedGroupListActivity extends AppCompatActivity {
-    private ListView lv;
-    private SupervisedGroupListViewModel supervisedGroupListViewModel;
-    private LiveData<List<Group>> listOfGroupsLiveData;
+    private ListView supervisedGroups;
+    private static SupervisedGroupListViewModel supervisedGroupListViewModel;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_supervised_group_list);
-        supervisedGroupListViewModel = new ViewModelProvider(this).get(SupervisedGroupListViewModel.class);
         initialize();
     }
 
     protected void initialize() {
         try {
-            FirebaseUser currentUser = supervisedGroupListViewModel.getCurrentUser();
-            if (currentUser != null) {
-                listOfGroupsLiveData = supervisedGroupListViewModel.getSupervisedGroupList();
-
-                listOfGroupsLiveData.observe(this, groups -> {
-                    ArrayList<String> arrayGroupName = new ArrayList<>();
-                    for (Group group : groups) {
-                        arrayGroupName.add(group.getName());
-                    }
-                    ArrayAdapter<String> arrayAdapter = new ArrayAdapter<>(getBaseContext(), android.R.layout.simple_list_item_1, arrayGroupName);
-                    lv = findViewById(R.id.listView_supervisedGroup);
-                    lv.setAdapter(arrayAdapter);
-                    // argument position gives the index of item which is clicked
-                    lv.setOnItemClickListener((arg0, v, position, arg3) -> {
+            supervisedGroupListViewModel = new ViewModelProvider(this).get(SupervisedGroupListViewModel.class);
+            LiveData<List<Group>> listOfGroupsLiveData = supervisedGroupListViewModel.getSupervisedGroupList();
+
+            listOfGroupsLiveData.observe(this, supervisedGroups -> {
+                if (supervisedGroups != null) {
+                    ArrayList<Group> groupsArrayList = new ArrayList<>(supervisedGroups);
+                    SupervisedGroupAdaptor adapter = new SupervisedGroupAdaptor(getBaseContext(), groupsArrayList);
+
+                    this.supervisedGroups = findViewById(R.id.supervised_groups);
+                    this.supervisedGroups.setAdapter(adapter);
+                    this.supervisedGroups.setOnItemClickListener((arg0, v, position, arg3) -> {
                         Intent intent = new Intent(getBaseContext(), SupervisedGroupPageActivity.class);
                         startActivity(intent);
                     });
-                });
-            }
-        } catch (Exception e) {
+                } else
+                    this.noGroupAlert().show();
+            });
+        } catch (
+                Exception e) {
             e.printStackTrace();
         }
     }
+
+
+    private static class SupervisedGroupAdaptor extends ArrayAdapter<Group> {
```

Comment:
You don't need static for this class.

Check the comment above.

Reply:
Got it! Thanks for pointing it out!